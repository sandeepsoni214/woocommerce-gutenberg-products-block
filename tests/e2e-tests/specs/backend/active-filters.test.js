/**
 * External dependencies
 */
import {
	insertBlock,
	getEditedPostContent,
	createNewPost,
	getAllBlocks,
	switchUserToAdmin,
} from '@wordpress/e2e-test-utils';

describe( 'Active Product Filters Block', () => {
	beforeEach( async () => {
		await switchUserToAdmin();
		await createNewPost();
	} );

	it( 'can be created', async () => {
		await insertBlock( 'Active Product Filters' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can only be inserted once', async () => {
		await insertBlock( 'Active Product Filters' );
		expect( await getAllBlocks() ).toHaveLength( 1 );

		await insertBlock( 'Active Product Filters' );
		expect( await getAllBlocks() ).toHaveLength( 1 );
	} );

	it( 'allows title can be manipulated', async () => {
		await insertBlock( 'Active Product Filters' );
		await expect( page ).toFill(
			'.wp-block[data-type="woocommerce/active-filters"] textarea.wc-block-component-title',
			'New Title'
		);
		await page.click(
			'.components-toolbar button[aria-label="Heading 6"]'
		);
		await expect(
			page
		).toMatchElement(
			'.wp-block[data-type="woocommerce/active-filters"] h6 textarea',
			{ text: 'New Title' }
		);

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Display style to change', async () => {
		await insertBlock( 'Active Product Filters' );
		await page.click( 'button[aria-label="Display Style: Chips"]' );
		await expect( page ).toMatchElement(
			'.wc-block-active-filters__list.wc-block-active-filters__list--chips'
		);
	} );
} );
