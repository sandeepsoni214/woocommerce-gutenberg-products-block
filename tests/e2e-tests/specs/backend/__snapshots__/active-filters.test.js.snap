// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Product Filters Block allows title can be manipulated 1`] = `
"<!-- wp:woocommerce/active-filters {\\"heading\\":\\"New Title\\",\\"headingLevel\\":6} -->
<div class=\\"wp-block-woocommerce-active-filters is-loading\\" data-display-style=\\"list\\" data-heading=\\"New Title\\" data-heading-level=\\"6\\"><span aria-hidden=\\"true\\" class=\\"wc-block-active-product-filters__placeholder\\"></span></div>
<!-- /wp:woocommerce/active-filters -->"
`;

exports[`Active Product Filters Block can be created 1`] = `
"<!-- wp:woocommerce/active-filters -->
<div class=\\"wp-block-woocommerce-active-filters is-loading\\" data-display-style=\\"list\\" data-heading=\\"Active filters\\" data-heading-level=\\"3\\"><span aria-hidden=\\"true\\" class=\\"wc-block-active-product-filters__placeholder\\"></span></div>
<!-- /wp:woocommerce/active-filters -->"
`;
